### PR TITLE
 [items] feature/CAES-131: replaced secret object with encrypted string

### PR DESCRIPTION
**Pull request by ipopov4xxi**

### CODE REVIEW

File: src/Migrations/2018/12/Version20181218102406.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+final class Version20181218102406 extends AbstractMigration
+{
+    public function up(Schema $schema): void
+    {
+        $this->abortIf('postgresql' !== $this->connection->getDatabasePlatform()->getName(), 'Migration can only be executed safely on \'postgresql\'.');
+
+        $this->addSql('ALTER TABLE post ALTER secret TYPE VARCHAR(65525)');
```
Comment (aburov-4xxi) [inline src/Migrations/2018/12/Version20181218102406.php]: How we can encrypt file? this length is very small for files. Why don't you use text type?

File: src/Migrations/2018/12/Version20181218102406.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        $this->addSql('ALTER TABLE post ALTER secret DROP DEFAULT');
+    }
+
+    public function down(Schema $schema): void
+    {
+        $this->abortIf('postgresql' !== $this->connection->getDatabasePlatform()->getName(), 'Migration can only be executed safely on \'postgresql\'.');
+
+        $this->addSql('ALTER TABLE post ALTER secret TYPE JSONB');
```
Comment (aburov-4xxi) [inline src/Migrations/2018/12/Version20181218102406.php]: extra SQL

File: src/Migrations/2018/12/Version20181218102406.php (hunk 1 above)
Comment (ipopov4xxi) [reply src/Migrations/2018/12/Version20181218102406.php]: Fixed

File: src/Migrations/2018/12/Version20181218102406.php (hunk 2 above)
Comment (ipopov4xxi) [reply src/Migrations/2018/12/Version20181218102406.php]: Fixed

File: src/Entity/Post.php [hunk 3]
```diff
@@ -129,17 +129,17 @@ public function setParentList(Directory $parentList)
...
     {
         return $this->secret;
     }
 
     /**
-     * @param array|null $secret
+     * @param string|null $secret
      */
-    public function setSecret(array $secret)
+    public function setSecret(?string $secret)
```
Comment (aburov-4xxi) [inline src/Entity/Post.php]: not null

Comment (ipopov4xxi) [reply src/Entity/Post.php]: Fixed

**Pull request merged**